### ISSUE TITLE
Wiring up using AddControllers() doesn't work. 

### ISSUE THREAD

**Issue by phillip-haydon** [FluentValidation/FluentValidation#1647]
**Which version of FluentValidation are you using?**
Please specify which version of FluentValidation you're using. 

FluentValidation.AspNetCore: 9.5.1

**Which version of ASP.NET are you using?**
If you are using FluentValidation with ASP.NET, please provide the version of ASP.NET that you're using (eg .NET Core 2.1, NET Core 3.1 etc)

.NET 5

**Describe the issue that you're having**
A clear and concise description of the issue that you're having, steps to reproduce, and sample code to reproduce the issue. 

Building a small API, I'm registering FluentValidation like so:

```
            services.AddControllers()
                    .AddFluentValidation(x => x.RegisterValidatorsFromAssemblyContaining<RegisterValidator>());

...

            app.UseEndpoints(endpoints =>
            {
                endpoints.MapControllers();
            });
```

The validators are never executed. 

If I changed `AddControllers` to `AddMvc` it all works totally fine. 

https://fizzylogic.nl/2020/12/18/upgrade-your-validation-logic-in-asp-net-core-with-fluentvalidation

I was following this blog post.

Unsure if this is broken in .NET 5 or unsupported? I haven't tried in 3.1

**Comment by JeremySkinner** [comment]
I believe this should work fine with both AddControllers and AddMvc. Can you provide a small sample project that repel rides the issue?

**Comment by phillip-haydon** [comment]
I started trying to setup a project to reproduce this, so created a new project and it just worked... So I went to my main project and it didn't work...

I pulled all my code into the new project, and it worked...

I renamed the new project as the old one, deleted the old folder and replaced it. And it works. Then I do a compare, and it says 0 changes... 

So I wonder if I had an old assembly lingering around in the `bin` directory that was resolved when I replaced the project. 

Sorry to waste your time :(
